src/gateway/routers/axelar/Router.sol

```diff
@@ -19,24 +20,47 @@ interface AggregatorLike {
     function handle(bytes memory message) external;
 }
 
+interface AxelarGasServiceLike {
+    function payNativeGasForContractCall(
+        address sender,
+        string calldata destinationChain,
+        string calldata destinationAddress,
+        bytes calldata payload,
+        address refundAddress
+    ) external payable;
+}
+
 /// @title  Axelar Router
 /// @notice Routing contract that integrates with an Axelar Gateway
-contract AxelarRouter is IRouter {
+
+contract AxelarRouter is IRouter, Auth {
     string public constant CENTRIFUGE_ID = "centrifuge";
     bytes32 public constant CENTRIFUGE_ID_HASH = keccak256(bytes("centrifuge"));
     bytes32 public constant CENTRIFUGE_ADDRESS_HASH = keccak256(bytes("0x7369626CEF070000000000000000000000000000"));
     string public constant CENTRIFUGE_AXELAR_EXECUTABLE = "0xc1757c6A0563E37048869A342dF0651b9F267e41";
 
     AggregatorLike public immutable aggregator;
     AxelarGatewayLike public immutable axelarGateway;
+    AxelarGasServiceLike public immutable axelarGasService;
 
-    constructor(address aggregator_, address axelarGateway_) {
+    /// @dev This value is in AXELAR fees in ETH ( wei )
```

Comment:
Um.... this is a variable

Reply:
Yeah fair point but that is actually something that we still need to decide. Quite a few others are using the same interface pattern also for variables. See e.g. https://github.com/morpho-org/morpho-blue/blob/main/src/Morpho.sol#L53

I'm a bit 50/50 on whether we should also adopt this

Reply:
Well I can argue about this specific case.
Owner is a generic variable that is used in most of the contract. It is not protocol / implementation specific .
In our case this is a very Axelar specific variable. It doesn't live outside this implementation.

Even though this variable per se is axelar specific, what we could do is actually make it more generic and add the `getter` as part of the interface ( What's been done in morpho ). In Axelar it's a constant but in other implementations it could be a calculated value.